plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse_test.go

```diff
@@ -0,0 +1,288 @@
+	var tests = []struct {
+		res    string
+		status int
+		subID  string
+	}{
+		{res: "{\"status\":200}", status: 200, subID: "subID"},
+		{res: "{\"status\":401}", status: 401, subID: ""},
+	}
+
+	for _, test := range tests {
+		t.Run(test.res, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(test.status)
+				_, err := w.Write([]byte(test.res))
+				require.NoError(t, err)
+			}))
+			defer server.Close()
+			subs := make([]Subscription, 0)
+			subs = append(subs, Subscription{
+				Nodes: []string{"plc/app/Application/sym/PLC_PRG/counter",
+					"plc/app/Application/sym/PLC_PRG/counterReverse"},
+			},
+			)
+			s := &CtrlXDataLayerSSE{
+				connection:   &http.Client{},
+				url:          server.URL,
+				Username:     "user",
+				Password:     "password",
+				Subscription: subs,
+				tokenManager: tokenManager{url: server.URL, username: "user", password: "password", connection: &http.Client{}, log: testutil.Logger{}},
+				Log:          testutil.Logger{},
+			}
+			subID, err := s.createSubscription(&subs[0])
+
+			if test.subID != "" && err == nil {
+				require.NotEmpty(t, subID)
+			} else {
+				require.Empty(t, subID)
+			}
+		})
+	}
+}
+
+func newServer(t *testing.T) *httptest.Server {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/identity-manager/api/v2/auth/token", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("{\"access_token\": \"eyJhbGciOiJIU.xxx.xxx\", \"token_type\":\"Bearer\"}"))
+		require.NoError(t, err)
+	}))
+	mux.HandleFunc("/identity-manager/api/v2/auth/token/validity", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("{\"valid\": \"true\"}"))
+		require.NoError(t, err)
+	}))
+	mux.HandleFunc(path+"/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodGet {
+			w.WriteHeader(http.StatusOK)
+			_, err := w.Write([]byte("event: update\n"))
+			require.NoError(t, err)
+			_, err = w.Write([]byte("id: 12345\n"))
+			require.NoError(t, err)
+			if getMultiEntries() {
+				_, err = w.Write([]byte("data: {\n"))
+				require.NoError(t, err)
+				_, err = w.Write([]byte("data: \"node\":\"motion/axisX/ipo/pos\", \"timestamp\":132669450604571037,\"type\":\"double\",\"value\":44.0\n"))
+				require.NoError(t, err)
+				_, err = w.Write([]byte("data: }\n"))
+				require.NoError(t, err)
+			} else {
+				_, err = w.Write([]byte("data: {\"node\":\"motion/axisX/ipo/pos\", \"timestamp\":132669450604571037,\"type\":\"double\",\"value\":43.0}\n"))
+				require.NoError(t, err)
+			}
+			_, err = w.Write([]byte("\n"))
+			require.NoError(t, err)
+		}
+	}))
+	mux.HandleFunc(path, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusCreated)
+		_, err := w.Write([]byte("201 created"))
+		require.NoError(t, err)
+	}))
+	return httptest.NewServer(mux)
+}
```

Comment:
The whole testing can benefit from using [existing SSE server implementations](https://pkg.go.dev/github.com/r3labs/sse)...

Reply:
As written above, the server is not spec compliant and for example rejects sse-clients that do not contain proprietary query parameters in the request.

Reply:
Please see my comment above. I think we cannot have a SSE server/client implementation as part of this plugin. This is certainly beyond the scope of telegraf and not a good separation of concerns.